### PR TITLE
[MRG] proposal for validator

### PR DESCRIPTION
**Pull request by JoseAlanis**
in relation to #4 
- adds a WIP version of the validator for the input directory

### CODE REVIEW

File: eegprep/utils/utils.py [hunk 1]
```diff
@@ -1,6 +1,118 @@
 
 
-def validate_input_dir(input_dir):
+
+def validate_sourcedata(path, source_type, pattern='sub-\\d+'):
+    """
+    This function validates the "sourcedata/" directory provided by user to
+    see if it's contents are consistent with BIDS.
```
Comment (PeerHerholz) [inline eegprep/utils/utils.py]: I'm not sure if the term `BIDS` should be used, as this proposed validator operates on the `sourcedata/` in order to check if everything is ok for a later conversion to `BIDS`, no?

File: eegprep/utils/utils.py [hunk 2]
```diff
@@ -1,6 +1,118 @@
...
+                                  string=os.path.basename(file)):
+                        valid_file_names.append(True)
+                    else:
+                        valid_file_names.append(False)
+
+            if len(set(valid_file_names)) > 1:
+                bad_files = np.where(np.logical_not(valid_file_names))[0]
+                bad_files = [data_files[bf] for bf in bad_files]
+
+                warnings.warn('Some file names are not BIDS conform.')
```
Comment (PeerHerholz) [inline eegprep/utils/utils.py]: Same as above.

**Review by PeerHerholz — COMMENTED** [review]
LGTM except a general term thingy...Please see comments.

File: eegprep/utils/utils.py (hunk 1 above)
Comment (JoseAlanis) [reply eegprep/utils/utils.py]: Ok, yes, I see your point. However, BIDS does have specific rules (or suggestions) for `sourcedata/` doesn’t it?

Comment (PeerHerholz) [reply eegprep/utils/utils.py]: It does, as you can check/evaluate here: https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#source-vs-raw-vs-derived-data. However, we would need to discuss how this applies to our case.

Comment (JoseAlanis) [reply eegprep/utils/utils.py]: Right, right, let’s kick it out for now. We can always com back and adapt the warning messages if necessary

File: eegprep/utils/utils.py (hunk 2 above)
Comment (JoseAlanis) [reply eegprep/utils/utils.py]: check

**Comment by JoseAlanis** [comment]
> LGTM except a general term thingy...Please see comments.

Took care of the comments. Ready for merging if everything ok.

**Comment by PeerHerholz** [comment]
Great, thx. I'll merge and close.

**Pull request merged**